test/test_torch.py

```diff
@@ -8035,45 +8035,43 @@ def test_complex_flip(self, device, dtype):
         for i in range(1, rand_dim):
             # Check all combinations of `i` axis.
             for flip_dim in combinations(range(rand_dim), i):
-                data = torch.randn(*shape, dtype=dtype).tolist()
+                data = torch.randn(*shape, device=device, dtype=dtype)
                 torch_fn = partial(torch.flip, dims=flip_dim)
                 np_fn = partial(np.flip, axis=flip_dim)
-                self.compare_with_numpy(torch_fn, np_fn, data, device, dtype)
+                self.compare_with_numpy(torch_fn, np_fn, data)
 
-    def _test_fliplr_flipud(self, funcs, min_dim, max_dim, device, dtype):
+    def _test_fliplr_flipud(self, torch_fn, np_fn, min_dim, max_dim, device, dtype):
         for dim in range(min_dim, max_dim + 1):
             shape = self._rand_shape(dim, 5, 10)
             # Randomly scale the input
-            data = (torch.randn(*shape).to(dtype) * random.randint(50, 100)).tolist()
-            # Use _np_compare_func as it copies the output of np_func,
-            # which takes care of negative strides if present.
-            torch_fn, np_fn = funcs
-            self.compare_with_numpy(torch_fn, np_fn, data, device, dtype)
+            if dtype.is_floating_point or dtype.is_complex:
```

Comment:
so this ignored `device` altogether? Nice.

Reply:
No. Historically it would create a list from a tensor and then compare_with_numpy would put that list into a tensor on the appropriate device.

The updated test just has fewer steps to get to the same place.